docs/backend/optimization.md

```diff
@@ -0,0 +1,156 @@
+    author = models.ForeignKey(Author, on_delete=models.CASCADE)
+    genres = models.ManyToManyField(Genre)
+    published_at = models.DateTimeField()
+
+class Publisher(models.Model):
+    name = models.CharField(max_length=300)
+    books = models.ManyToManyField(Book)
+
+# Without prefetch_related, this hits the database with multiple queries.
+publisher = Publisher.objects.get(name="BigBooks") # Hits the database.
+related_books = publisher.books.all()  # Hits the database again to get the related books.
+
+# Using prefetch_related()...
+publisher = Publisher.objects.prefetch_related("books__author", "books__genres").get(name="BigBooks")
+related_books = publisher.books.all()  # Doesn't hit the database for authors and genres, as they are included in the initial query.
+
+```
+
+### Using `.only()`
+
+[Django's `.only()`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#only) method is a powerful tool for optimizing queries by allowing you to fetch only specific fields from the database, reducing the amount of data retrieved. This can be particularly useful when you are interested in a subset of fields and want to minimize the overhead associated with fetching unnecessary data.
+
+Consider the following example with a Book model:
+
+```python
+from django.db import models
+
+class Book(models.Model):
+    title = models.CharField(max_length=100)
+    author = models.CharField(max_length=100)
+    published_at = models.DateTimeField()
+    page_count = models.IntegerField()
+    genre = models.CharField(max_length=50)
+
+
+# Without using .only(), this fetches all fields from the database.
+all_books = Book.objects.all()
+
+# Using .only() to fetch only specific fields - title and author.
+selected_fields_books = Book.objects.only('title', 'author').all()
+```
+
+This optimization becomes more noticeable when dealing with larger datasets or when only a subset of fields is required for a particular operation. By utilizing .only(), you can enhance the performance of your Django application by minimizing unnecessary data transfer from the database.
+
+### Optimizing Django-Ninja Schema
+
+When defining Pydantic schemas for Django-Ninja, explicitly include only the fields that are essential for processing or need to be included in the response. Avoid adding unnecessary fields to minimize the data transmitted over the network and improve the overall efficiency of your API.
+
+In the following example, the `OperationListOut` schema is defined to include only the essential fields from the `Operation` model to be used on the client side.(Operations Table)
+
+```python
+from ninja import ModelSchema
+
+class OperationListOut(ModelSchema):
+    operator: str = Field(..., alias="operator.legal_name")
+    bc_obps_regulated_operation: Optional[str] = Field(None, alias="bc_obps_regulated_operation.id")
+
+    class Config:
+        model = Operation
+        model_fields = ['id', 'name', 'bcghg_id', 'submission_date', 'status']
```

Comment:
If I didn't know about aliases, this example would confuse me because I wouldn't know if it was lines 150-151 or the `model_fields` that was the fields we wanted. Maybe use a simpler example, or link to the  ninja docs?

Reply:
Removed those two confusing lines.